tls/s2n_client_hello.c

```diff
@@ -83,6 +91,9 @@ int s2n_client_hello_recv(struct s2n_connection *conn)
     if (conn->actual_protocol_version == S2N_TLS12) {
         conn->secure.conn_hash_alg = S2N_HASH_SHA1;
     }
+    if (s2n_is_in_fips_mode()) {
+        conn->secure.conn_hash_alg = S2N_HASH_SHA256;
```

Comment:
FYI I also changed this to:
```
if (FIPS) {
    SHA256
} else if (TLS 1.2) {
    SHA1
}
```
This was done to show that we are explicitly overriding the TLS 1.2 default when in FIPS mode.